packages/demo/bridge/services/identity/eventhandler/stellar_payment_handler.go

```diff
@@ -0,0 +1,137 @@
+package eventhandler
+
+import (
+	"strconv"
+
+	"github.com/rate-engineering/rate3-monorepo/packages/demo/bridge/services/identity/db"
+	"github.com/rate-engineering/rate3-monorepo/packages/demo/bridge/support/stellar"
+	"github.com/stellar/go/clients/horizon"
+	"go.uber.org/zap"
+	"gopkg.in/guregu/null.v3"
+)
+
+func (h *StellarPaymentHandler) Receive(account *db.StellarAccount, payment *horizon.Payment) {
+	switch payment.Type {
+	case "create_account":
+		h.handleAccountCreation(account, payment)
+	case "payment":
+		h.handleIncomingPayment(account, payment)
+	}
+}
+
+func (h *StellarPaymentHandler) handleAccountCreation(account *db.StellarAccount, payment *horizon.Payment) {
+	if account.Cursor.Valid {
+		return
+	}
+
+	rowsAffected, err := h.DB.SetStellarAccountCursor(account.ID, payment.PagingToken)
+	if err != nil || rowsAffected == 0 {
+		h.Logger.Error("Unable to set account cursor",
+			zap.Error(err),
+			zap.Int64("Account ID", account.ID),
+			zap.String("Account address", account.Address),
+			zap.String("Cursor", payment.PagingToken),
+		)
+	}
+}
+
+func (h *StellarPaymentHandler) handleIncomingPayment(account *db.StellarAccount, payment *horizon.Payment) {
+	ethAddress, memo, memoType := h.parseMemo(payment)
+
+	receivedPayment := db.ReceivedPayment{
+		ToAccountID: account.ID,
+		FromAccount: payment.From,
+		TxHash:      payment.TransactionHash,
+		PagingToken: payment.PagingToken,
+		MemoType:    memoType,
+		Memo:        memo,
+	}
+	if id, err := strconv.Atoi(payment.ID); err == nil {
+		receivedPayment.PaymentID = int64(id)
+	}
+
+	switch payment.Type {
+	case "create_account":
+		receivedPayment.FromAccount = payment.Funder
+	}
+
+	if !memo.Valid {
```

Comment:
Is there a check on valid ethereum address? based on checksum

Reply:
No, ethereum address will only have checksum if they are represented in **string** form with capitalization. The memo that we are expecting here is a hexadecimal in bytes form.